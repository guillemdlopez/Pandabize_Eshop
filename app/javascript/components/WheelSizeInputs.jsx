import React from "react";
import PropTypes from "prop-types";

const WheelSizeInputs = ({
  wheelSizes,
  handleInputChange,
  fetchAssociations,
  setPrice,
  setPriceWheel,
}) => {
  return (
    <ul className="block-list-center">
      {wheelSizes?.map((customization) => (
        <li key={customization.id} id="wheel-sizes">
          <label htmlFor={customization.name}>
            {customization.name}: <strong>{customization.value}</strong>
          </label>
          <br />
          <input
            type="radio"
            name="wheelSize"
            onClick={(e) => {
              handleInputChange(e);
              fetchAssociations(customization.id, e);
              setPrice(customization.price);
              setPriceWheel(customization.price);
            }}
            value={customization.id}
            className="ml-2"
          />
        </li>
      ))}
    </ul>
  );
};

WheelSizeInputs.propTypes = {
  wheelSizes: PropTypes.array,
  handleInputChange: PropTypes.func.isRequired,
  fetchAssociations: PropTypes.func.isRequired,
  setPriceWheel: PropTypes.func.isRequired,
};

export default WheelSizeInputs;
