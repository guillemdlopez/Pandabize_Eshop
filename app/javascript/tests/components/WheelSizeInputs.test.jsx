import { shallow } from "enzyme";
import React from "react";
import WheelSizeInputs from "../../components/WheelSizeInputs";

describe("Testing the <WheelSizeInputs/> component", () => {
  const wheelSizeInputsProps = {
    wheelSizes: [
      {
        id: 146,
        name: "Wheel size",
        value: "15 inches",
        price: 9.23,
        bicycle_id: 26,
      },
      {
        id: 148,
        name: "Wheel size",
        value: "16 inches",
        price: 6.15,
        bicycle_id: 26,
      },
      {
        id: 149,
        name: "Wheel size",
        value: "17 inches",
        price: 8.42,
        bicycle_id: 26,
      },
    ],
    handleInputChange: jest.fn(),
    fetchAssociations: jest.fn(),
    setPriceWheel: jest.fn(),
  };
  const wrapper = shallow(<WheelSizeInputs {...wheelSizeInputsProps} />);

  test("should render it correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should display a label and an input for each wheel size", () => {
    const { wheelSizes } = wheelSizeInputsProps;
    const sizes = wheelSizes.map((size) => size.value);
    const [size15, size16, size17] = sizes;

    expect(wrapper.find("label").at(0).text()).toBe(`Wheel size: ${size15}`);
    expect(wrapper.find("label").at(1).text()).toBe(`Wheel size: ${size16}`);
    expect(wrapper.find("label").at(2).text()).toBe(`Wheel size: ${size17}`);

    expect(wrapper.find("input").length).toBe(sizes.length);
  });
});
