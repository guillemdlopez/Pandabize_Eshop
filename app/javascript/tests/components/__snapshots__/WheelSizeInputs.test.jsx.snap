// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <WheelSizeInputs/> component should render it correctly 1`] = `
<ul
  className="block-list-center"
>
  <li
    id="wheel-sizes"
    key="146"
  >
    <label
      htmlFor="Wheel size"
    >
      Wheel size
      : 
      <strong>
        15 inches
      </strong>
    </label>
    <br />
    <input
      className="ml-2"
      name="wheelSize"
      onClick={[Function]}
      type="radio"
      value={146}
    />
  </li>
  <li
    id="wheel-sizes"
    key="148"
  >
    <label
      htmlFor="Wheel size"
    >
      Wheel size
      : 
      <strong>
        16 inches
      </strong>
    </label>
    <br />
    <input
      className="ml-2"
      name="wheelSize"
      onClick={[Function]}
      type="radio"
      value={148}
    />
  </li>
  <li
    id="wheel-sizes"
    key="149"
  >
    <label
      htmlFor="Wheel size"
    >
      Wheel size
      : 
      <strong>
        17 inches
      </strong>
    </label>
    <br />
    <input
      className="ml-2"
      name="wheelSize"
      onClick={[Function]}
      type="radio"
      value={149}
    />
  </li>
</ul>
`;
